azurelinuxagent/ga/exthandlers.py

```diff
@@ -1403,13 +1396,13 @@ def download(self):
             # Validate package signature
             if should_validate_ext_signature:
                 try:
-                    # TODO: set 'failure_log_level' to ERROR when signature validation is enforced.
-                    validate_signature(package_file, self.ext_handler.encoded_signature, package_full_name=self.get_full_name(), failure_log_level=logger.LogLevel.WARNING)
+                    validate_signature(package_file, self.ext_handler.encoded_signature, package_full_name=self.get_full_name(), failure_log_level=validation_failure_log_level)
                     signature_validated = True
-                except SignatureValidationError:
-                    # validate_signature() only raises SignatureValidationError and sends logs/telemetry for the error, so do nothing here.
-                    # TODO: Raise error once signature validation is enforced.
-                    pass
+                except SignatureValidationError as ex:
+                    # validate_signature() only raises SignatureValidationError and sends logs/telemetry for the error, so no need to log here.
+                    if enforce_signature:
```

Comment:
Discussed offline: currently the value of 'enforce_signature' is being set by Policy. We need to separate 2 kinds of errors: an extension not being signed, and, when the extension is signed, an error during signature validation. Policy only controls the former. We need to separate those 2 kinds of errors into different flags/settings.